File: Bidapp Demo App - ObjectiveC/bidappPlayground/BannerDelegate.m
//
//  BannerDelegate.m
//  bidappPlayground
//
//  Copyright 2023 bidapp. All rights reserved.
//

#import "BannerDelegate.h"
#import "tools.h"
#import <bidapp/BIDAdInfo.h>

@implementation BannerDelegate

- (void)bannerDidLoad:(BIDBannerView *)banner adInfo:(BIDAdInfo *)adInfo
{
    NSLog(@"[%@][%@] bannerDidLoad: %d", sessionIdShort(adInfo.showSessionId), adInfo.waterfallId, adInfo.networkId);
    
    if (!banner.isAdDisplayed)
    {
        [banner refreshAd];
    }
}

- (void)bannerDidDisplay:(BIDBannerView *)banner adInfo:(BIDAdInfo *)adInfo
{
	NSLog(@"[%@][%@] bannerDidDisplay: %d", sessionIdShort(adInfo.showSessionId), adInfo.waterfallId, adInfo.networkId);
}

- (void)bannerDidFailToDisplay:(BIDBannerView *)banner adInfo:(BIDAdInfo *)adInfo error:(NSError *)error
{
	NSLog(@"[%@][%@] bannerDidFailToDisplay: %d ERROR: %@", sessionIdShort(adInfo.showSessionId), adInfo.waterfallId, adInfo.networkId, error.localizedDescription ? error.localizedDescription : error);
}

- (void)bannerDidClick:(BIDBannerView *)banner adInfo:(BIDAdInfo *)adInfo
{
	NSLog(@"[%@][%@] bannerDidClicked: %d", sessionIdShort(adInfo.showSessionId), adInfo.waterfallId, adInfo.networkId);
}

- (void)allNetworksFailedToDisplayAdInBanner:(BIDBannerView *)banner
{
    NSLog(@"allNetworksFailedToDisplayAdInBanner");
}

@end

File: Bidapp Demo App - Swift/bidappPlayground/BannerDelegate.swift
//
//  BannerDelegate.swift
//  bidappPlayground
//
//  Copyright 2023 bidapp. All rights reserved.
//

import UIKit
import bidapp

class BannerDelegate : NSObject, BIDBannerViewDelegate {
    
    func bannerDidLoad(_ banner: BIDBannerView, adInfo: BIDAdInfo) {
        print("[\(String(describing: adInfo.showSessionId))][\(String(describing: adInfo.waterfallId))] bannerDidLoad: \(adInfo.networkId) [\(adInfo)]")
        
        if !banner.isAdDisplayed() {
            banner.refreshAd()
        }
    }
    
    func bannerDidDisplay(_ banner: BIDBannerView, adInfo: BIDAdInfo) {
        print("[\(String(describing: adInfo.showSessionId))][\(String(describing: adInfo.waterfallId))] bannerDidDisplay: \(adInfo.networkId) [\(adInfo)]")
    }
    
    func bannerDidFail(toDisplay banner: BIDBannerView, adInfo: BIDAdInfo, error: Error) {
        print("[\(String(describing: adInfo.showSessionId))][\(String(describing: adInfo.waterfallId))] bannerDidFailToDisplay: \(adInfo.networkId) [\(adInfo)] ERROR: \(error.localizedDescription)")
    }
    
    func bannerDidClick(_ banner: BIDBannerView, adInfo: BIDAdInfo) {
        print("[\(String(describing: adInfo.showSessionId))][\(String(describing: adInfo.waterfallId))] bannerDidClick: \(adInfo.networkId) [\(adInfo)]")
    }
    
    func allNetworksFailedToDisplayAd(inBanner banner: BIDBannerView) {
        print("allNetworksFailedToDisplayAd")
    }
}
